src/esm_parser/yaml_to_dict.py

```diff
@@ -733,7 +730,7 @@ def load(self, stream):
         mapping_with_tuple_prov = super().load(stream)[0]
 
         config, provenance = self._extract_dict(mapping_with_tuple_prov)
-
+        self.env_variables = self.constructor.env_variables
```

Comment:
I couldn't find out, why `self.Constructor.env_variables` does not work. This attribute is set in the `EnvironmentConstructor` class. `self.env_variables = self.Constructor.env_variables` complaines that 

`type object 'ProvenanceConstructor' has no attribute 'env_variables'`

`self.Constructor` exists, but does not have attribute `env_variables`.